src/Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests/PullDeliveryTests.cs

```diff
@@ -10,6 +10,7 @@ namespace Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests;
 
 public class PullDeliveryTests(ITestOutputHelper testOutputHelper)
 {
+    private const int EmulatorPort = 6504;
```

Comment:
TestContainers should assign an external port randomly, so you don't have to fix one.
If you have multiple tests, it should start multiple containers with different external ports.

You can get the randomly assigned port using `container.GetMappedPublicPort(6500 /* internal port */)`

Reply:
I am currently testing around both this one and the new PushDeliveryTests. There's an issue that is only showing in the CI and I'm not sure what it is, I suspected a potential race condition so I tried to explicitly split the ports without much success. I'm still trying to figure out what is breaking in the CI, as these pass when ran locally.

Reply:
The testing to find the CI failure root cause is completed, this should now be resolved